src/main/java/com/google/javascript/cl2dts/DeclarationGenerator.java

```diff
@@ -282,30 +308,33 @@ private void walkScope(TypedVar symbol, final String namespace, boolean isDefaul
       }
       emit(getUnqualifiedName(symbol));
 
-      Function<ObjectType, String> relativeName = new Function<ObjectType, String>() {
-        @Override public String apply(ObjectType input) {
-          return !input.getDisplayName().startsWith(namespace) ? input.getDisplayName()
-              : input.getDisplayName().substring(namespace.length() + 1);
-        }
-      };
+      if (type.hasAnyTemplateTypes()) {
+        emit("<");
+        emit(FluentIterable.from(ftype.getTemplateTypeMap().getTemplateKeys()).transform(new Function<JSType, String>() {
```

Comment:
Yeah, I enjoyed writing the functional style but I'm not married to it. You're very naughty to suggest an if body without braces just to justify the smaller line count, however :smile: 
changed to iterative